### PR TITLE
[release] fix torch lightning gpu tests.

### LINKED ISSUE #38036: [release] torch lightning gpu tests are failing
https://buildkite.com/ray-project/release-tests-branch/builds/2018

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Why are these changes needed?
Replace `validation_epoch_end ` with `on_ validation_epoch_end `.

## Related issue number
Closes #38036

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/lightning_tests/workloads/lightning_test_utils.py [hunk 1]
```diff
@@ -34,11 +35,17 @@ def validation_step(self, val_batch, batch_idx):
...
-    def validation_epoch_end(self, outputs):
-        avg_loss = torch.stack([x["val_loss"] for x in outputs]).mean()
-        avg_acc = torch.stack([x["val_accuracy"] for x in outputs]).mean()
+    def on_validation_epoch_end(self):
+        avg_loss = torch.stack(
+            [x["val_loss"] for x in self.validation_step_outputs]
+        ).mean()
+        avg_acc = torch.stack(
+            [x["val_accuracy"] for x in self.validation_step_outputs]
+        ).mean()
```
Comment (woshiyyya) [inline release/lightning_tests/workloads/lightning_test_utils.py:48]:
```suggestion
        ).mean()
        self.validation_step_outputs.clear()
```

**Review by woshiyyya — CHANGES_REQUESTED** [review]
Thanks @xwjiang2010! LGTM. 

Just one minor change: clear the validation outputs at the end of validation epoch.

**Comment by xwjiang2010** [comment]
> Thanks @xwjiang2010! LGTM.
> 
> Just one minor change: clear the validation outputs at the end of validation epoch.

Ah right!
This is for correctness right? As the data structure is being used for validation that happens in different epochs. So we want to have a fresh start for another new epoch.

**Comment by woshiyyya** [comment]
@xwjiang2010  Yes that's right:)

**Review by woshiyyya — APPROVED** [review]
Thank you @xwjiang2010 ~

**Pull request merged**